### PR TITLE
fix ignoreIfSupported for non-team V2 signatures

### PR DESCRIPTION
**Pull request by maxtaco**
- test cases for {ignored,not-ignored}X{stubbed,unstubbed}

### CODE REVIEW

**Comment by maxtaco** [comment]
Companion PR: https://github.com/keybase/keybase-test-vectors/pull/18

File: go/libkb/chain_link_v2.go [hunk 1]
```diff
@@ -236,6 +264,9 @@ func DecodeStubbedOuterLinkV2(b64encoded string) (*OuterLinkV2WithMetadata, erro
 	if err != nil {
 		return nil, err
 	}
+	if !ol.IgnoreIfUnsupported.True() && !ol.LinkType.IsSupportedType() {
+		return nil, ChainLinkStubbedUnsupportedError{fmt.Sprintf("Stubbed link with type %d is unknown and not marked with IgnoreIfUnsupported", ol.LinkType)}
```
[the remark this answers is not in the document]
Comment (maxtaco) [reply go/libkb/chain_link_v2.go]: Really?

**Comment by maxtaco** [comment]
@mlsteele PTAL

File: go/libkb/chain_link_v2.go [hunk 2]
```diff
@@ -63,7 +63,33 @@ func (t SigchainV2Type) NeedsSignature() bool {
 	case SigchainV2TypeTrack, SigchainV2TypeUntrack, SigchainV2TypeAnnouncement:
 		return false
 	default:
+		// Unsupported types don't need signatures. Otherwise we can't
+		// make code forwards-compatible.
+		return t.IsSupportedType()
+	}
+}
+
+func (t SigchainV2Type) IsSupportedType() bool {
```
Comment (mlsteele) [inline go/libkb/chain_link_v2.go]: This function kind of seems like a trap. Would two separate `IsSupportedUserType` and `IsSupportedTeamType`  work?

Comment (maxtaco) [reply go/libkb/chain_link_v2.go]: ok sure

File: go/libkb/chain_link_v2.go [hunk 3]
```diff
@@ -210,7 +238,7 @@ func MakeSigchainV2OuterSig(
 		Curr:                currLinkID,
 		LinkType:            v2LinkType,
```
Comment (mlsteele) [inline go/libkb/chain_link_v2.go]: so this ends up as `SigchainV2TypeNone` for unsupported links?

**Comment by maxtaco** [comment]
merging over CI, the build is fine, it worked on my machine

**Pull request merged**